### ISSUE TITLE
Switch to buffer loses focus for the page with many windows

### ISSUE THREAD

**Issue by mariari** [atlas-engineer/nyxt#1463]
**Describe the bug**

When one has many windows, and tries to search for a page to switch to, they may have their input interrupted and have it focus on the page.

This happens when nyxt switches to the window that is on the other window. Now your commands work on the page itself, and going back to the mini-buffer either requires a click, or doing a command that brings up the mini-buffer then escaping it (nyxt remembers to focus the mini-buffer after this).

**Precise recipe to reproduce the issue**

Make multiple windows with C-x 5 2. Have github or some other website on the other monitor, on the monitor without the git site, try to type git, if it comes up and switches it from the other monitor, your focus should be on the page itself.

**Information**
- OS name+version:
  - Arch Linux 
- Graphics card and driver:
- Desktop environment / Window manager name+version:
  - Xmonad/Stumpwm 
- How you installed Nyxt (Guix pack, package manager, build from source):
  - Source 
- Information from command copy-system-information:

  If you can't run copy-system-information, try `nyxt --system-information` from
  a shell.  It this still does not work, please provide the following:
  - Nyxt version (from =M-x nyxt-version= or =nyxt --version=):
  - Lisp implementation/version (if built from source):
  - Kernel name+version:
  - WebKitGTK+ | QtWebEngine version:

**Output when started from a shell**

**Comment by Ambrevar** [comment]
To clarify: you mean that when typing in the prompt buffer, it sometimes loses focus?

If so, I can reproduce.

**Comment by mariari** [comment]
Correct, yes.

**Comment by Ambrevar** [comment]
@aartaka @jmercouris Any idea how to force-focus a GTK widget on window switch?

**Comment by jmercouris** [comment]
Yes, i believe I already utilize this in the codebase.

```
gtk:gtk-widget-grab-focus
```

then you need to wrap some sort of signal around window changing I believe.

**Comment by jmercouris** [comment]
There is a bigger problem though, the user may expect to click on a widget in another window and have it focused, if you instead jump the focus to the prompt buffer, then the user will be annoyed.

I suggest:

A binding to swap the focus between the prompt buffer and the main buffer.

**Comment by Ambrevar** [comment]
> then you need to wrap some sort of signal around window changing I
> believe.

Yes, this is what I would need.  I can't find this signal however.
Maybe `::activate-focus`?
https://docs.gtk.org/gtk4/class.Window.html#signals
Or maybe `::move-focus`?

I don't think the user would be annoyed because the automatic focus
would only occur on window change.

Also a keybinding defeats the purpose of this fix because the point is
to have the prompt buffer _automatically_ focused.

**Comment by jmercouris** [comment]
The conclusion is the whole input dispatch system must be rewritten, properly- without pointless indirection.
